lfs.c

```diff
@@ -1400,8 +1400,17 @@ static int lfs_dir_drop(lfs_t *lfs, lfs_mdir_t *dir, lfs_mdir_t *tail) {
 
     // steal tail
     lfs_pair_tole32(tail->tail);
-    err = lfs_dir_commit(lfs, dir, LFS_MKATTRS(
-            {LFS_MKTAG(LFS_TYPE_TAIL + tail->split, 0x3ff, 8), tail->tail}));
+    {
+        struct lfs_mattr __lfs_mattr__[]=
+        {
+            {
+                (((lfs_tag_t)(LFS_TYPE_TAIL + tail->split) << 20) | ((lfs_tag_t)(0x3ff) << 10) | (lfs_tag_t)(8)), 
+                tail->tail
+            },
+        };
+        int __attrcount__=sizeof(__lfs_mattr__)/sizeof(struct lfs_mattr);
+        err = lfs_dir_commit(lfs, dir, __lfs_mattr__,__attrcount__);
```

Comment:
Sorry, this change adds too much noise and defeats the API of the internal lfs_dir_commit function. Is there an alternative?

Reply:
no .the tasking compiler is a professional compiler in the field of automotive electronics. This compiler does not support high C language standard(such as : c14)

Reply:
What warning does cctc output for this line? LFS_MKATTRS just passes a point to a stack allocated array of static size, which shouldn't be too strange.

littlefs is should be strictly c99, otherwise useful features such as inline, arbitrary variable declarations, uint32_t, wouldn't be available.